gobblin-runtime/src/main/java/org/apache/gobblin/runtime/fork/Fork.java

```diff
@@ -229,7 +229,7 @@ public void consumeRecordStream(RecordStreamWithMetadata<D, S> stream)
         }
       }, e -> {
           logger.error("Failed to process record.", e);
-          throw(new RuntimeException(e));
+          verifyAndSetForkState(ForkState.RUNNING, ForkState.FAILED);
```

Comment:
Does the writer need to be closed or will it be taken care of as part of shutdown sequence?

Reply:
writer close is handled in the doFinal() part. It will be executed when the subscription is finished either by error or complete.